physical/file.go

```diff
@@ -121,19 +137,43 @@ func (b *FileBackend) Get(k string) (*Entry, error) {
 }
 
 func (b *FileBackend) Put(entry *Entry) error {
-	path, key := b.path(entry.Key)
+	if entry == nil {
+		return fmt.Errorf("nil entry")
+	}
+
+	var err error
+	basePath, fileName := b.path(entry.Key)
+
+	// New storage entries will have their file names base64 URL encoded. If a
+	// file with a non-encoded file name exists, it indicates that this is an
+	// update operation. To avoid duplication of storage entries, delete the
+	// old entry first.
```

Comment:
I think the old entry shouldn't be deleted first; if there is an error writing the new entry then you've just had data loss.

If we prefer the base64'd version, having an old version lying around isn't really a problem unless you delete it and the old version isn't deleted.

My suggestion is that this deletion should actually be in a deferred function that is invoked iff the stat against a previous non-base64'd value is successful. This would also have to take place with the lock, so it would need to be after the lock acquisition (and also, defers are LIFO so the defer statement itself must be after the deferred unlock).

In Delete, you could help avoid any issues by also stating the non-base64 version, and deleting whichever version(s) are found.